### ISSUE TITLE
Login to CLI using Azure AD Federated Login

### ISSUE THREAD

**Issue by atbtechgit** [lastpass/lastpass-cli#519]
Is there the ability to login to the CLI using Azure AD Federated Login?

**Comment by TWFpa2Vs** [comment]
please bring this feature, i feel crippled without lastpass cli :(

**Comment by josephtallison** [comment]
Agreed, this is a necessary feature for me.

**Comment by internetionals** [comment]
We are currently unable to use LastPass for storing many of our service secrets because of this and instead rely on https://passwordstore.org for that half of our secret storing and sharing needs.

Perhaps something like the `az login` method that the Azure CLI uses could be added relatively easy? Here you get an URL and a shortish authentication token that you use (if you're running `az login` on the same device as your webbrowser it opens it automatically) and this way it can match the authentication request to the client and go throught the normal web based federated login. I presume this creates an oauth token that it relays back to the original requesting cli that has been polling/waiting on the result from the server.

**Comment by jakubigla** [comment]
I'm an enterprise customer and this feature is also necessary. Maybe we could introduce and idea of a token?

**Comment by jerri** [comment]
We just switched to federated login and I am sad to see that the cli application is not working anymore. It would be amazing if this would be added into the cli app.

**Comment by sickrager** [comment]
> Is there the ability to login to the CLI using Azure AD Federated Login?

I agree with the requirement. I think this is a necessary option. Please integrate this feature.

**Comment by sys-adm90** [comment]
> Is there the ability to login to the CLI using Azure AD Federated Login?

I think also this is a must have feature.

**Comment by Natfan** [comment]
Arguably a requirement for enterprise users. I must say that I am very surprised by the absolute lack of any support for Windows/AzureAD users of this product.

**Comment by ShadowXVII** [comment]
The lacklusture response from LastPass support:

> Thank you very much for the information;I was working with my next level of support, and they inform me that This type of request must go through the GitHub channel as this is an open-source program. 
> 
> Unfortunately, as LastPass we do not have any updated information for CLI; I really do apologize.
